src/incubator/ChipsInput/index.tsx

```diff
@@ -0,0 +1,224 @@
+
+const removeIcon = require('./assets/xSmall.png');
+
+export enum ChipsInputChangeReason {
+  Added = 'added',
+  Removed = 'removed'
+}
+
+export type ChipsInputProps = Omit<TextFieldProps, 'ref'> & {
+  chips?: ChipProps[];
+  defaultChipProps?: ChipProps;
+  onChange?: (chips: ChipProps[], changeReason: ChipsInputChangeReason, updatedChip: ChipProps) => void;
+  /**
+   * Maximum chips
+   */
+  maxChips?: number;
+};
+
+const ChipsInput = (props: ChipsInputProps, refToForward: React.Ref<any>) => {
+  const fieldRef = useCombinedRefs(refToForward);
+  const {chips = [], defaultChipProps, leadingAccessory, onChange, fieldStyle, maxChips, ...others} = props;
+  const [markedForRemoval, setMarkedForRemoval] = useState<number | undefined>(undefined);
+  const fieldValue = useRef(others.value);
+
+  const addChip = useCallback(() => {
+    const reachedMaximum = maxChips && chips?.length >= maxChips;
+    if (fieldValue.current && !reachedMaximum) {
+      const newChip = {label: fieldValue.current};
+      onChange?.([...chips, newChip], ChipsInputChangeReason.Added, newChip);
+      setMarkedForRemoval(undefined);
+      // @ts-expect-error
+      fieldRef.current.clear();
+      fieldValue.current = '';
+    }
+  }, [onChange, chips, maxChips]);
+
+  const removeMarkedChip = useCallback(() => {
+    if (!isUndefined(markedForRemoval)) {
+      const removedChip = chips?.splice(markedForRemoval, 1);
+      onChange?.([...chips], ChipsInputChangeReason.Removed, removedChip?.[0]);
+      setMarkedForRemoval(undefined);
+    }
+  }, [chips, markedForRemoval, onChange]);
+
+  const onChipPress = useCallback(({customValue: index}) => {
+    const selectedChip = chips[index];
+    selectedChip?.onPress?.();
+
+    setMarkedForRemoval(index);
+  },
+  [chips]);
+
+  const onChangeText = useCallback(value => {
+    fieldValue.current = value;
+    props.onChangeText?.(value);
+
+    if (!isUndefined(markedForRemoval)) {
+      setMarkedForRemoval(undefined);
+    }
+  },
+  [props.onChangeText, markedForRemoval]);
+
+  const onKeyPress = useCallback((event: NativeSyntheticEvent<TextInputKeyPressEventData>) => {
+    props.onKeyPress?.(event);
+    const keyCode = event?.nativeEvent?.key;
+    const pressedBackspace = keyCode === Constants.backspaceKey;
+
+    if (pressedBackspace && !fieldValue.current) {
+      if (isUndefined(markedForRemoval) || markedForRemoval !== chips.length - 1) {
+        setMarkedForRemoval(chips.length - 1);
+      } else {
+        removeMarkedChip();
+      }
+    }
+  },
+  [chips, props.onKeyPress, markedForRemoval, removeMarkedChip]);
+
+  const chipList = useMemo(() => {
+    return (
+      <>
+        {leadingAccessory}
+        {map(chips, (chip, index) => {
+          const isMarkedForRemoval = index === markedForRemoval;
+          return (
+            <Chip
+              key={index}
+              customValue={index}
+              // resetSpacings
+              // paddingH-s2
```

Comment:
Can be removed?

Reply:
I kept it for now so I'll have a quick reference in case there is something missing while we're migrating. 
I added a TODO